### PR TITLE
Refine depencies + Odoo 18 support

### PR DESCRIPTION
**Pull request by sbidoul**

### CODE REVIEW

**Comment by sbidoul** [comment]
@qgroulard @marielejeune @samirGuesmi for 17 & 18

**Review by marielejeune — APPROVED** [review]

File: src/pyproject.toml.jinja [hunk 1]
```diff
@@ -52,12 +52,28 @@ dependencies = [
...
+  "pypdf2<3",
+  "werkzeug==3.0.*",
+  "setuptools<71", # https://github.com/pypa/setuptools/issues/4516
+  "xlrd<2",  # for xlsx support
+  {% elif odoo_series == "18.0" %}
+  "packaging",  # https://github.com/odoo/odoo/pull/181768
+  "pypdf2<3",
+  "werkzeug==3.0.*",
+  "xlrd<2",  # for xlsx support
+  {% endif %}
```
Comment (qgroulard) [inline src/pyproject.toml.jinja:76]: I don't understand where they come from, but on APL I have phonenumbers and packaging in the requirements without adding them here.

Comment (sbidoul) [reply src/pyproject.toml.jinja:76]: @qgroulard packaging is a dependency of `manifestoo-core`, so it likely comes from there. `phonenumber` I don't know. Can you do a `pip-df tree` to check where it comes from?

Comment (qgroulard) [reply src/pyproject.toml.jinja]: Indeed `packaging` comes with `manifestoo-core`.
And I found `phonenumber` there: https://github.com/acsone/enterprise/blob/18.0/whatsapp/__manifest__.py

Comment (sbidoul) [reply src/pyproject.toml.jinja]: Ok, so we can keep it as is, as both packaging and phonenumbers are optional dependencies of Odoo CE.

**Review by qgroulard — APPROVED** [review]

**Pull request merged**